File: internal/extract/testdata/typescript/basic.ts
export class User {
  name: string;

  constructor(name: string) {
    this.name = name;
  }

  greet(): string {
    return "hello";
  }

  static build(): User {
    return new User("foo");
  }
}

export interface Greeter {
  greet(): string;
}

export type Name = string;

export enum Color {
  Red = "red",
  Blue = "blue",
}

export const VERSION = "1.0";

export const handler = (req: Request) => {
  return new Response();
};

export function helper() {
  return 42;
}

File: internal/extract/testdata/c/basic.c
#include <stdio.h>
#include <stdlib.h>

struct Point {
    int x;
    int y;
};

enum Color {
    RED,
    GREEN,
    BLUE
};

void greet(const char* name) {
    printf("Hello, %s\n", name);
}

int add(int a, int b) {
    return a + b;
}

int main() {
    greet("world");
    int result = add(1, 2);
    printf("result: %d\n", result);
    return 0;
}
